### PR TITLE
FOGL-1010: Unit test for core/api/scheduler.py

### PR DESCRIPTION
**Pull request by Singhal-Vaibhav**
- Test results

`77 passed, 1 xfailed in 3.28 seconds`

- Code coverage:

`Coverage for /FogLAMP/python/foglamp/services/core/api/scheduler.py : 97% 
322 statements   312 run 10 missing 0 excluded`

**Note:** _The 3% code which is not covered is the unreachable code that needs to be refactored._

### CODE REVIEW

File: python/foglamp/services/core/api/scheduler.py [hunk 1]
```diff
@@ -75,9 +75,6 @@
 
     scheduled_process_name = request.match_info.get('scheduled_process_name', None)
 
-    if not scheduled_process_name:
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/scheduler.py]: @Singhal-Vaibhav why this is removed, what if `?scheduled_process_name=`

File: python/foglamp/services/core/api/scheduler.py [hunk 2]
```diff
@@ -296,9 +293,6 @@ def _extract_args(data, curr_value):
     try:
         schedule_id = request.match_info.get('schedule_id', None)
 
-        if not schedule_id:
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/scheduler.py]: why this is removed, what if `?schedule_id=` i.e. value is empty string

File: python/foglamp/services/core/api/scheduler.py [hunk 3]
```diff
@@ -334,9 +328,6 @@ def _extract_args(data, curr_value):
...
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/scheduler.py]: as above

File: python/foglamp/services/core/api/scheduler.py [hunk 4]
```diff
@@ -366,9 +357,6 @@ def _extract_args(data, curr_value):
...
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/scheduler.py]: as above

File: tests/unit/python/foglamp/services/core/api/test_scheduler_api.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+            process = ScheduledProcess()
+            process.name = "foo"
+            process.script = "bar"
+            processes.append(process)
+            return processes
+
+        server.Server.scheduler = Scheduler(None, None)
+        with patch.object(server.Server.scheduler, 'get_scheduled_processes', return_value=mock_coro()):
+            resp = await client.get('/foglamp/schedule/process')
+            result = await resp.text()
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/services/core/api/test_scheduler_api.py]: resp.status?

File: tests/unit/python/foglamp/services/core/api/test_scheduler_api.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+from foglamp.services.core.scheduler.exceptions import *
+
+__author__ = "Vaibhav Singhal"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("api", "core")
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/services/core/api/test_scheduler_api.py]: @pytest.allure.story("core", "api", "schedule")

File: tests/unit/python/foglamp/services/core/api/test_scheduler_api.py [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+            schedule.time = None
+            schedule.day = None
+            return schedule
+
+        server.Server.scheduler = Scheduler(None, None)
+        with patch.object(server.Server.scheduler, 'get_schedule', return_value=mock_coro()):
+            resp = await client.get('/foglamp/schedule/{}'.format(self._random_uuid))
+            assert 200 == resp.status
+            result = await resp.text()
+            json_response = json.loads(result)
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/services/core/api/test_scheduler_api.py]: result is actual json_response, you mean response_dict = json.loads(result) ?

Comment (Singhal-Vaibhav) [reply tests/unit/python/foglamp/services/core/api/test_scheduler_api.py]:
```
print(type(result))
 <class 'str'>
```

File: tests/unit/python/foglamp/services/core/api/test_scheduler_api.py [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+        response = storage_return
+        with patch.object(connect, 'get_storage', return_value=storage_client_mock):
+            with patch.object(storage_client_mock, 'query_tbl_with_payload', return_value=response):
+                with patch.object(server.Server.scheduler, 'get_schedule',
+                                  return_value=mock_coro()) as patch_get_schedule:
+                    resp = await client.put('/foglamp/schedule/{}'.format(self._random_uuid), data=json.dumps(request_data))
+                    assert response_code == resp.status
+                    assert error_message == resp.reason
+
+    async def test_delete_schedule(self, client):
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/services/core/api/test_scheduler_api.py]: add 1 more test `test_delete_enabled_schedule` it should return something like 409 (enabled schedule can not be deleted)

**Pull request merged**